salt/client/ssh/__init__.py

```diff
@@ -374,7 +374,11 @@ def _expand_target(self):
                 roster_data = self.__parsed_rosters[roster_filename]
                 if not isinstance(roster_data, bool):
                     for host_id in roster_data:
-                        if hostname in [host_id, roster_data[host_id].get("host")]:
+                        if isinstance(roster_data[host_id], salt.utils.odict.OrderedDict):
```

Comment:
I've given this a bit more thought and I think it might be better to do something like this instead:

```
diff --git a/salt/client/ssh/__init__.py b/salt/client/ssh/__init__.py
index 987cf73388..dea8a60de4 100644
--- a/salt/client/ssh/__init__.py
+++ b/salt/client/ssh/__init__.py
@@ -374,9 +374,9 @@ class SSH:
                 roster_data = self.__parsed_rosters[roster_filename]
                 if not isinstance(roster_data, bool):
                     for host_id in roster_data:
-                        if isinstance(roster_data[host_id], salt.utils.odict.OrderedDict):
+                        try:
                             roster_host = roster_data[host_id].get("host")
-                        else:
+                        except AttributeError:
                             roster_host = roster_data[host_id]
                         if hostname in [host_id, roster_host]:
                             if hostname != self.opts["tgt"]:
```

The reason being is someone can use a different renderer other than yaml and it will not return an OrderedDict. I tested this by using the python render in the salt-ssh roster and it returned a dictionary. This will also fix the failing test `test_expand_target_ip_address`